apps/webstore/src/app/pages/auth/checkout/checkout.component.ts

```diff
@@ -236,13 +249,81 @@ export class CheckoutComponent implements OnInit {
   }
 
   submitCheckout() {
+    // check checkout form and if terms are accepted
+    if (!this.checkoutForm.valid || !this.isTermsAndConditionsAccepted) {
+      this.checkoutInvalid = true;
+      this.checkInputFields();
+      return;
+    }
+
+    // check if billing address is the same
+    if (!this.billingAddressIsTheSame) {
+      if (!this.billingAddressForm.valid) {
+        this.billingAddressInvalid = true;
+        this.checkBillingFields();
+        return;
+      }
+    }
+
+    this.checkoutInvalid = false;
     if (this.isLoggedIn) {
       this.createOrder();
     } else {
       this.createOrderWithNewUser();
     }
   }
 
+  checkInputFields() {
```

Comment:
Should be form specific naming since this is only checking checkoutForm.

Reply:
Changed this. Will see the updated names of checkShippingInputFields and checkBillingInputFields in next commit.